src/test/kotlin/com/kafkakotlin/demo/users/UserRepositoryTest.kt

```diff
@@ -0,0 +1,87 @@
+package com.kafkakotlin.demo.users
+
+import com.kafkakotlin.demo.kafka.producer.KafkaProducer
+import com.kafkakotlin.demo.kafka.statestorequery.StateStoreQuery
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import io.mockk.slot
+import org.apache.kafka.streams.KeyValue
+import org.apache.kafka.streams.state.KeyValueIterator
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.kafka.config.StreamsBuilderFactoryBean
+import org.springframework.web.client.RestTemplate
+
+internal class UserRepositoryTest {
+    private val kafkaProducer = mockk<KafkaProducer>()
+    private val store = mockk<StateStoreQuery>()
+    private val streamsBuilderFactoryBean = mockk<StreamsBuilderFactoryBean>()
+    private val restTemplate = mockk<RestTemplate>()
+    private val kafkaProperties = mockk<KafkaProperties>()
+
+    private val properties = mapOf("application.server" to "localhost:9999")
+    private val testUser = User("steve", "steve@example.com", "super-strong-password-1")
+
+    private val keyValueIterator = mockk<KeyValueIterator<String, User>>()
+    private lateinit var underTest: UserRepository
+
+    @BeforeEach
+    fun setup() {
+        every { kafkaProperties.streams.properties } returns properties
+
+        underTest = UserRepository(kafkaProducer, store, streamsBuilderFactoryBean, restTemplate, kafkaProperties)
+    }
+
+    @Test
+    fun `should pass given User to kafkaProducer when creating a user`() {
+        val slot = slot<User>()
```

Comment:
have a look at `mockk` `verify`, we tend to use it more than argument captors/slots